GVRf/Framework/src/org/gearvrf/GVRViewManager.java

```diff
@@ -667,7 +675,9 @@ public void beforeDrawEyes() {
 
             doMemoryManagementAndPerFrameCallbacks();
 
-            mScript.onStep();
+            GVRViewManager.this.getEventManager().sendEvent(
```

Comment:
This is a pretty slow operation - 12-14ms most of times on my S6 (even for a script without onStep). I can see some things could be cached but I doubt it can ever be very fast considering what is involved.

I suggest moving the script execution off the GL thread. From the script perspective the threading shouldn't be apparent anyway. This may mean a lot of work on the GVRF side - we could leave it for later as long as it is ensured that no overhead is incurred for the use-case where no scripts are involved.